core/src/test/java/org/apache/calcite/test/JdbcTest.java

```diff
@@ -3925,6 +3926,28 @@ public void checkOrderBy(final boolean desc,
         .returnsUnordered("empid=150; name=Sebastian");
   }
 
+  /** Test case for <a href="https://issues.apache.org/jira/browse/CALCITE-6904">
+   * [CALCITE-6904] IS_NOT_DISTINCT_FROM is converted error in ElementerableJoinRule</a>. */
+  @Disabled("CALCITE-6904")
+  @Test void testIntersectToSemiJoin() {
+    final String sql = ""
+        + "select \"commission\" from \"hr\".\"emps\"\n"
+        + "intersect\n"
+        + "select \"commission\" from \"hr\".\"emps\" where \"empid\">=150";
+    CalciteAssert.hr()
+        .query(sql)
+        .withHook(Hook.PLANNER, (Consumer<RelOptPlanner>)
+            planner -> {
+              planner.removeRule(CoreRules.INTERSECT_TO_DISTINCT);
```

Comment:
can you just remove this new rule here as well and not disable the test?

Reply:
This is due to jira 6904, which has nothing to do with the new rule. The purpose of adding this case is to test IntersectToSemiJoin. If this rule is removed, this test may not be needed.